client/asset/eth/eth.go

```diff
@@ -668,12 +619,166 @@ func (eth *ExchangeWallet) unlockFunds(coins asset.Coins) error {
 	return nil
 }
 
-// Swap sends the swaps in a single transaction. The Receipts returned can be
-// used to refund a failed transaction. The Input coins are manually unlocked
-// because they're not auto-unlocked by the wallet and therefore inaccurately
-// included as part of the locked balance despite being spent.
-func (*ExchangeWallet) Swap(swaps *asset.Swaps) ([]asset.Receipt, asset.Coin, uint64, error) {
-	return nil, nil, 0, asset.ErrNotImplemented
+// swapReceipt implements the asset.Receipt interface for ETH.
+type swapReceipt struct {
+	txHash     common.Hash
+	secretHash [32]byte
+	// expiration and value can be determined with a blockchain
+	// lookup, but we cache these values to avoid this.
+	expiration time.Time
+	value      uint64
+}
+
+// Expiration returns the time after which the contract can be
+// refunded.
+func (r *swapReceipt) Expiration() time.Time {
+	return r.expiration
+}
+
+// Coin returns the coin used to fund the swap.
+func (r *swapReceipt) Coin() asset.Coin {
+	return &coin{
+		value: r.value,
+		id:    dex.Bytes(r.txHash[:]),
+	}
+}
+
+// Contract returns the swap's secret hash.
+func (r *swapReceipt) Contract() dex.Bytes {
+	return dex.Bytes(r.secretHash[:])
+}
+
+// String returns a string representation of the swapReceipt.
+func (r *swapReceipt) String() string {
+	return fmt.Sprintf("tx hash: %x, secret hash: %x", r.txHash, r.secretHash)
+}
+
+// SignedRefund returns an empty byte array. ETH does not support a pre-signed
+// redeem script becuase the nonce needed in the transaction can not be previously
+// determined. We will need to come up with a different way to support manual
+// refunds in ETH.
+func (*swapReceipt) SignedRefund() dex.Bytes {
+	return dex.Bytes{}
+}
+
+var _ asset.Receipt = (*swapReceipt)(nil)
+
+// Swap sends the swaps in a single transaction. The Receipts returned can
+// be used to refund a failed transaction. The fees used returned are the max
```

Comment:
Maybe the Receipts comment is wrong for eth. They cannot be used to refund, because the the signature will need to include the next nonce for the account to be accepted by the network. Right? And those must be in ascending order incrementing by one for each mined transaction.

Reply:
Yes that's right, the comment is wrong.